### PR TITLE
StringValue improvements

### PR DESCRIPTION
**Pull request by bronek89**

### CODE REVIEW

File: src/StringValue.php [hunk 1]
```diff
@@ -125,6 +125,11 @@ public function matchPatterns(string $pattern);
 
     public function isMatching(string $pattern): bool;
 
+    /**
+     * @param string|StringValue $pattern
+     */
+    public function isStartingWith($pattern): bool;
```
Comment (dazet) [inline src/StringValue.php]: Maybe it should have more popular name `startsWith` like in Java, JS, Python, Rust... Laravel helpers :P

Comment (bronek89) [reply src/StringValue.php]: I've followed convention from method above - `isMatching` :)

File: src/PlainString.php [hunk 2]
```diff
@@ -189,4 +208,24 @@ public function isEmpty(): bool
...
+            return $value;
+        }
+
+        if ($value instanceof StringValue) {
+            return $value->toString();
+        }
+
+        if (\is_scalar($value)) {
+            return (string)$value;
+        }
```
Comment (dlakomski) [inline src/PlainString.php:289]: Maybe check if `__toString()` exists?

Comment (bronek89) [reply src/PlainString.php:289]: Added :)

**Review by dazet — APPROVED** [review]

**Pull request merged**